### ISSUE TITLE
Review file upload/drag and drop UX

### ISSUE THREAD

**Issue by michael-coker** [syndesisio/syndesis#3402]
## This is a...

<pre><code>
[ ] Feature request
[ ] Regression (a behavior that used to work and stopped working in a new release)
[x] Bug report  
[ ] Documentation issue or request
</code></pre>

## The problem
We have 3 instances of a file upload that support drag 'n drop in fuse online. Here are the discrepancies:

* Integration importer
   * Auto imports the integration without a review step
   * No error message on invalid file type
   * Error message shown above (as an alert) and inside file upload area
* Extension importer
   * File validation errors shown above (as an alert) file upload area, and general upload errors shown in file upload area
   * on successful upload, the file name is shown beside (not combined with) the success status message
* API Connector importer
   * Need to review validation/error handling outside of invalid file type error.

We need to review the pattern we're using as a whole, and fix any consistencies or discrepancies between them so they all provide a similar UX.

For reference
* [PatternFly File Upload pattern](https://www.patternfly.org/pattern-library/forms-and-controls/file-upload/)
* [Dropzone.js](https://www.dropzonejs.com/)
* [ng-file-upload](https://angular-file-upload.appspot.com/)

cc @amysueg @sjcox-rh @seanforyou23

**Comment by michael-coker** [comment]
related https://github.com/syndesisio/syndesis/issues/3371

**Comment by gashcrumb** [comment]
I think that'd be ideal, however we need the backend updated to support that.  Right now there's one endpoint that accepts the file(s) and it automatically imports it.  Want to raise an issue and tag it for the server folks?

**Comment by paoloantinori** [comment]
hi guys? was this just a UX activity? do you consider this done?

**Comment by gashcrumb** [comment]
No, this is blocked on #3420 really, as importing an integration behaves differently, the file upload is directly applied, rather than being staged for import, unlike extension or API client connector file uploads.

**Comment by dongniwang** [comment]
Here's the v1 of the File Upload UX design https://redhat.invisionapp.com/share/NFO7XHSA4TX, please review and provide comments in InVision. Thank you! 

@michael-coker @gashcrumb @TovaCohen @gaughan 
cc; @amysueg @sjcox-rh @seanforyou23

**Comment by TovaCohen** [comment]
This looks good to me. I think I noticed one typo, which I commented on. 
And this is a tiny nit, but there should be a period at the end of each instance of the following since each one is a complete sentence:

- Drag your file here or browse to upload
- Drag your files here or browse to upload
- Drop file here to upload
- File type must be .json

**Comment by gaughan** [comment]
LGTM! No period on the .json sentence tho, that would add confusion

**Comment by michael-coker** [comment]
Awesome @dongniwang! Left some comments, and fwiw I like option 1 for the import preview for multiple files. It's more consistent with the existing UI and it allows us to display as many fields as we want on the screen without a user having to click to view details of the import.
